tools/xsl/common.xsl

```diff
@@ -0,0 +1,164 @@
+<xsl:template name="header-path">
+<xsl:variable name="elem" select="str:tokenize(d:deviceType | d:serviceType | s:serviceType, ':')"/>
+<xsl:value-of select="concat($elem[2], '/', $elem[3], '/', $elem[4], $elem[5], '.h')"/>
+</xsl:template>
+
+<!-- Device or service type with version, e.g. RenderingControl1 -->
+<xsl:template name="control-class">
+<xsl:variable name="elem" select="str:tokenize(d:deviceType | d:serviceType | s:serviceType, ':')"/>
+<xsl:value-of select="concat($elem[4], $elem[5])"/>
+</xsl:template>
+
+<!-- Just the device or service type, no version, e.g. RenderingControl -->
+<xsl:template name="control-name"><xsl:call-template name="urn-type"/></xsl:template>
+
+<!-- Domain element of URN -->
+<xsl:template name="urn-domain"><xsl:value-of select="str:tokenize(d:deviceType | d:serviceType | s:serviceType, ':')[2]"/></xsl:template>
+
+<!-- Domain element of URN in C++ compatible format -->
+<xsl:template name="urn-domain-cpp">
+<xsl:variable name="elem" select="str:tokenize(d:deviceType | d:serviceType | s:serviceType, ':')"/>
+<xsl:value-of select="translate($elem[2], '.-', '__')"/>
+</xsl:template>
+
+<!-- Will be "device" or "service" -->
+<xsl:template name="urn-kind"><xsl:value-of select="str:tokenize(d:deviceType | d:serviceType | s:serviceType, ':')[3]"/></xsl:template>
+
+<!-- Type element of URN -->
+<xsl:template name="urn-type"><xsl:value-of select="str:tokenize(d:deviceType | d:serviceType | s:serviceType, ':')[4]"/></xsl:template>
+
+<!-- Version element of URN -->
+<xsl:template name="urn-version"><xsl:value-of select="str:tokenize(d:deviceType | d:serviceType | s:serviceType, ':')[5]"/></xsl:template>
+
+<!-- State variable type -->
+<xsl:template name="variable-type">
+	<xsl:param name="const"/>
+	<xsl:variable name="vartype" select="/s:scpd/s:serviceStateTable/s:stateVariable[s:name=current()/s:relatedStateVariable]/s:dataType"/>
+	<xsl:call-template name="cpp-vartype">
+		<xsl:with-param name="vartype" select="$vartype"/>
+		<xsl:with-param name="const" select="$const"/>
+	</xsl:call-template>
+</xsl:template>
+
+<!-- Name of action with weird prefixes removed -->
+<xsl:template name="action-name"><xsl:value-of select="str:replace(s:name, 'X_', '')"/></xsl:template>
+
+<!-- Name of variable/parameter with weird prefixes removed, for display -->
+<xsl:template name="varname"><xsl:value-of select="str:replace(s:name, 'X_', '')"/></xsl:template>
+
+<!-- Name of variable/parameter, for C++ use -->
+<xsl:template name="varname-cpp">v<xsl:call-template name="varname"/></xsl:template>
+
+<!-- Name of structure containing action result -->
+<xsl:template name="action-result"><xsl:call-template name="action-name"/>Result</xsl:template>
+
+<!-- Name of result Delegate type -->
+<xsl:template name="action-result-callback"><xsl:call-template name="action-result"/>Callback</xsl:template>
+
+<!-- Method declaration for service action -->
+<xsl:template name="action-method">action_<xsl:call-template name="action-name"/>(
```

Comment:
Maybe the names can be `actionSomeThing` instead of the mixed `action_SomeThing`?

Reply:
What were your concerns when adding the `action_` prefix. Why not just use lower-case-first-letter(methodName). Instead of `action_SomeThing` to have `someThing`?

Reply:
It would be nice to camelCase both action methods and variable names, but it does run the risk of clashing with an existing method. Same goes for types. The only reason for `action_*` vs `action*` is to highlight that these are UPnP action methods and not regular methods.

I'll push a commit with lower camel-case naming and see how things go!

Reply:
> I'll push a commit with lower camel-case naming and see how things go!

Great, thanks a lot!